cloudferrylib/os/network/neutron.py

```diff
@@ -857,10 +916,23 @@ def upload_subnets(self, networks, subnets):
                 continue
             tenant_id = \
                 self.identity_client.get_tenant_id_by_name(snet['tenant_name'])
+            if not tenant_id:
+                LOG.debug("Cannot get tenant_id for subnet {subnet}".format(
+                    subnet=snet.get("id")))
+                continue
             net_hash = \
                 self.get_res_hash_by_id(networks, snet['network_id'])
+            if not net_hash:
+                LOG.debug("Cannot get network info for subnet {subnet}".format(
+                    subnet=snet.get("id")))
+                continue
             network = \
                 self.get_res_by_hash(existing_nets, net_hash)
+            if not network:
```

Comment:
I guess we do not need this check, because we can not create a subnet without a network. It is mandatory requirement.